Allow the editing of startup schedules

ItsDoot opened the pull request:
# Objective

Fixes #10968 

## Solution

Pull startup schedules from a list of `ScheduleLabel`s in the same way the update schedules are handled.

---

## Changelog

- Added `MainScheduleOrder::startup_labels` to allow the editing of the startup schedule order.

## Migration Guide

- Added a new field to `MainScheduleOrder`, `startup_labels`, for editing the startup schedule order.

alice-i-cecile reviewed (approved):
Good! That's about as easy as I hoped it would be.

alice-i-cecile commented:
Oh, this is technically breaking since we've added a new field to a public struct.

ItsDoot commented:
> Oh, this is technically breaking since we've added a new field to a public struct. For all of the 0 people explicitly constructing this!

Added a simple migration guide 👍

andriyDev commented:
This might be another issue, but should we also add support for PreFixedUpdate/PostFixedUpdate? This same mechanism could be used for that too.

Someone in the discord was asking about PostFixedUpdate 😅

alice-i-cecile commented:
That's clever. Happy to see it done, here or elsewhere, but we shouldn't block this PR on that.

ItsDoot commented:
> This might be another issue, but should we also add support for PreFixedUpdate/PostFixedUpdate? This same mechanism could be used for that too.

Happy to look into it in a separate PR. :+1:
